testsupport/wait/host.go

```diff
@@ -1186,13 +1187,26 @@ func (a *HostAwaitility) WaitForNotificationWithName(t *testing.T, notificationN
 			return false, fmt.Errorf("notification found with name does not have the expected type")
 		}
 
-		return matchNotificationWaitCriterion([]toolchainv1alpha1.Notification{notification}, criteria...), nil
+		return matchNotificationWaitCriterion([]toolchainv1alpha1.Notification{*notification}, criteria...), nil
 	})
 	// no match found, print the diffs
 	if err != nil {
-		a.printNotificationWaitCriterionDiffs(t, []toolchainv1alpha1.Notification{notification}, criteria...)
+		a.printNotificationWaitCriterionDiffs(t, []toolchainv1alpha1.Notification{*notification}, criteria...)
 	}
-	return notification, err
+	return *notification, err
+}
+
+// WaitForNotificationToBeNotCreated waits and checks that notification is NOT created.
+func (a *HostAwaitility) WaitForNotificationToNotBeCreated(t *testing.T, notificationName string) error {
+	t.Logf("waiting to check notification with name '%s' is NOT created", notificationName)
+	err := wait.Poll(a.RetryInterval, 10*time.Second, func() (done bool, err error) {
+		notification := &toolchainv1alpha1.Notification{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: notificationName, Namespace: a.Namespace}, notification); err != nil {
+			return false, err
+		}
+		return true, err
+	})
+	return err
```

Comment:
this won't keep waiting for 10s again - it will technically return immediately. 
I had something like this in mind:
```suggestion
	t.Logf("waiting to check notification with name '%s' is NOT created", notificationName)
	notification := &toolchainv1alpha1.Notification{}
	err := wait.Poll(a.RetryInterval, 10*time.Second, func() (done bool, err error) {
		notification = &toolchainv1alpha1.Notification{}
		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: notificationName, Namespace: a.Namespace}, notification); err != nil {
			if errors.IsNotFound(err) {
				return false, nil
			}
			return false, err
		}
		return true, nil
	})
	if err == nil {
		return fmt.Errorf("notification '%s' was found, but it was expected to not be created/present: \n %v", notificationName, notification)
	}
	if err == wait.ErrWaitTimeout {
		return nil
	}
	return err
```

and then you can call it:
```
err = hostAwait.WaitForNotificationToNotBeCreated(t, "test-idler-stage-idled")
require.NoError(t, err)
```

so it's technically very similar to the other function `WaitForNotificationWithName`, we just change the expectations

Reply:
yes right i had tried this before but somehow ended up doing the other way round, updated it now